src/Reflex/Dynamic.hs

```diff
@@ -181,10 +182,26 @@ count e = holdDyn 0 =<< zipListWithEvent const (iterate (+1) 1) e
 toggle :: (Reflex t, MonadHold t m, MonadFix m) => Bool -> Event t a -> m (Dynamic t Bool)
 toggle = foldDyn (const not)
 
+-- | Switches to the new 'Event' whenever it receives one.  Switching occurs
+-- *after* the inner 'Event' fires - so if the 'Dynamic' changes and both the
+-- old and new inner Events fire simultaneously, the output will fire with the
+-- value of the *old* 'Event'.
+--
+-- Prefer this to 'switchPromptlyDyn' where possible. The lack of doing double
+-- work when the outer and (new) inner fires means this imposes fewer "timing
+-- requirements" and thus is far more easy to use without introducing fresh
+-- failure cases.
```

Comment:
Might want to also mention that it has better performance than the prompt version.